opensearchutil/bulk_indexer.go

```diff
@@ -150,8 +150,20 @@ type BulkIndexerResponseItem struct {
 		Type   string `json:"type"`
 		Reason string `json:"reason"`
 		Cause  struct {
-			Type   string `json:"type"`
-			Reason string `json:"reason"`
+			Type        string   `json:"type"`
+			Reason      string   `json:"reason"`
+			ScriptStack []string `json:"script_stack"`
+			Script      string   `json:"script"`
+			Lang        string   `json:"lang"`
+			Position    struct {
+				Offset int `json:"offset"`
+				Start  int `json:"start"`
+				End    int `json:"end"`
+			} `json:"position"`
+			Cause struct {
+				Type   string `json:"type"`
+				Reason string `json:"reason"`
+			} `json:"caused_by"`
```

Comment:
Are all this fields are non-optional if error exists?

Reply:
For the case I shared, yes they-re non-optional. I'm not sure about other errors cases.

Reply:
If we are not sure, we should make it optional too so that it will not break. Can you make those extra variables as pointer and add omitempty

Reply:
Done.